src/array.rs

```diff
@@ -24,15 +24,24 @@ impl<T: RuntimeType> Array<T> {
     /// Creates an array of the given length with default values.
     pub fn with_len(len: usize) -> Self {
         assert!(len < std::u32::MAX as usize);
```

Comment:
Unrelated to this pull request, but I need to ask: Why is this `<` rather than `<=`? It is my understanding, that `std::u32::MAX` is a valid argument to [CoTaskMemAlloc](https://docs.microsoft.com/en-us/windows/win32/api/combaseapi/nf-combaseapi-cotaskmemalloc).

Is there something I'm missing?

Reply:
@kennykerr I believe @tim-weis has a point here. Can you confirm?

Reply:
Technically yes. In practice arrays aren't used for large buffers because some language projections would have a real hard time with them (as they don't marshal them efficiently) and there are more efficient ways to share large buffers.

Reply:
Alright we can make the change just to be "more correct", but it's not a priority. PRs welcome 😄